### PR TITLE
fix failing test cases

### PR DESCRIPTION
**Pull request by FCanberk**
fixed 5 failing test cases under `class ProxyOfProxyTest(unittest.TestCase):`

set the `BASE_DIRECTORY` using `tempfile.gettempdir()` to ensure the folder exist on both windows and linux operating systems.

### CODE REVIEW

File: src/odemis/model/_core.py [hunk 1]
```diff
@@ -53,14 +55,15 @@
 INIT_TIMEOUT = 300  # s
 CALL_TIMEOUT = 30  # s
 
-# TODO needs a different value on Windows
-# TODO try a user temp directory if /var/run/odemisd doesn't exist (and cannot be created)
-BASE_DIRECTORY="/var/run/odemisd"
-BASE_GROUP="odemis" # user group that is allowed to access the backend
-
+# Set the base directory regardless of the os
+BASE_DIRECTORY = tempfile.gettempdir() + "/odemisd"
```
Comment (tepals) [inline src/odemis/model/_core.py:59]: to make it completely os independent you should use:
```
BASE_DIRECTORY = os.path.join(tempfile.gettempdir(), "odemisd")
```

File: src/odemis/model/_core.py [hunk 2]
```diff
@@ -19,6 +19,8 @@
 You should have received a copy of the GNU General Public License along with
 Odemis. If not, see http://www.gnu.org/licenses/.
 '''
+import tempfile
+
 from past.builtins import basestring
 import Pyro4
```
Comment (tepals) [inline src/odemis/model/_core.py:25]: If you want you could add another commit to update the import order according to the new coding guidelines and replace the `'''` with `"""` on line 2 and line 21

**Review by tepals — CHANGES_REQUESTED** [review]
few small extra comments

**Review by pieleric — CHANGES_REQUESTED** [review]

File: src/odemis/model/test/remote_test.py [hunk 3]
```diff
@@ -300,7 +300,7 @@ def test_dataflow_unsub(self):
 
         comp2.sub(comp.data)
         time.sleep(0.5)
-#         comp2.unsub()
+        comp2.unsub()
```
Comment (pieleric) [inline src/odemis/model/test/remote_test.py:303]: This should not be uncommented. It's to show where the unsub() call would be happening, but it's not happening and still things should work.... What you can do is to add a comment explaining that.

**Review by pieleric — CHANGES_REQUESTED** [review]

**Review by tepals — APPROVED** [review]

**Pull request merged**